gtsfm/bundle/bundle_adjustment.py

```diff
@@ -412,6 +423,63 @@ def evaluate(
 
         return ba_metrics
 
+    def _run_ba_with_profiling(
```

Comment:
nit: _run_ba_and_evaluate() might be a better name, since we compute other metrics too.

Reply:
Good idea, updated now to `_run_ba_and_evaluate()`